src/main/java/com/mraof/minestuck/data/CassetteSongsProvider.java

```diff
@@ -0,0 +1,177 @@
+		
+		addHitEffect(MSSoundEvents.JUKEBOX_SONG_DANCE_STAB_DANCE, .3F, new MobEffectInstance(MobEffects.DIG_SLOWDOWN, 600, 3), MSCassetteSongs.DANCE_STAB_DANCE.song());
+		addHitEffect(MSSoundEvents.JUKEBOX_SONG_EMISSARY_OF_DANCE, .1F, new MobEffectInstance(MobEffects.POISON, 200, 0), MSCassetteSongs.EMISSARY_OF_DANCE.song());
+		addSelfEffect(MSSoundEvents.JUKEBOX_SONG_RETRO_BATTLE_THEME, new MobEffectInstance(MobEffects.DIG_SPEED, 100, 2, false, false, false), MSCassetteSongs.RETRO_BATTLE_THEME.song());
+	}
+	
+	protected void addHitEffect(ResourceKey<JukeboxSong> song, MobEffectInstance effect, String name)
+	{
+		addHitEffect(song, 1, effect, name);
+	}
+	
+	protected void addHitEffect(ResourceKey<JukeboxSong> song, float chance, MobEffectInstance effect, String name)
+	{
+		add(song, chance, effect, true, name);
+	}
+	
+	protected void addHitEffect(ResourceKey<JukeboxSong> song, MobEffectInstance effect, ResourceLocation name)
+	{
+		addHitEffect(song, 1, effect, name);
+	}
+	
+	protected void addHitEffect(ResourceKey<JukeboxSong> song, float chance, MobEffectInstance effect, ResourceLocation name)
+	{
+		add(new CassetteSong(song, new EffectContainer(effect, chance, true)), name);
+	}
+	
+	protected void addSelfEffect(ResourceKey<JukeboxSong> song, MobEffectInstance effect, String name)
+	{
+		add(song, 1, effect, false, name);
+	}
+	
+	protected void addSelfEffect(ResourceKey<JukeboxSong> song, MobEffectInstance effect, ResourceLocation name)
+	{
+		add(new CassetteSong(song, new EffectContainer(effect, 1, false)), name);
+	}
+	
+	protected void add(ResourceKey<JukeboxSong> song, MobEffectInstance effect, boolean onHit, String name)
+	{
+		add(song, 1, effect, onHit, name);
+	}
+	
+	protected void add(ResourceKey<JukeboxSong> song, float chance, MobEffectInstance effect, boolean onHit, String name)
+	{
+		add(new CassetteSong(song, new EffectContainer(effect, chance, onHit)), name);
+	}
+	
+	protected void add(CassetteSong song, String name)
+	{
+		add(song, ResourceLocation.fromNamespaceAndPath(modid, name));
+	}
+	
+	protected void add(CassetteSong song, ResourceLocation name)
+	{
+		songs.put(name, song);
+	}
+	
+	@Override
+	public CompletableFuture<?> run(CachedOutput cache)
+	{
+		registerSongs();
+		
+		Path outputPath = output.getOutputFolder();
+		List<CompletableFuture<?>> futures = new ArrayList<>(songs.size());
+		
+		for(Map.Entry<ResourceLocation, CassetteSong> entry : songs.entrySet())
+		{
+			Path songPath = getPath(outputPath, entry.getKey());
+			JsonElement jsonData = CassetteSong.CODEC.encodeStart(JsonOps.INSTANCE, entry.getValue()).getOrThrow();
+			sortCures(jsonData);
+			futures.add(DataProvider.saveStable(cache, jsonData, songPath));
+		}
+		
+		return CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
+	}
+	
+	private static Path getPath(Path outputPath, ResourceLocation id)
+	{
+		return outputPath.resolve("data/" + id.getNamespace() + "/minestuck/cassette_songs/" + id.getPath() + ".json");
```

Comment:
Lets change the path from `cassette_songs` to `cassette_effects` or something similar